app/config/AppConfigModule.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+import com.google.inject.{AbstractModule, Provides, Singleton}
+
+class AppConfigModule extends AbstractModule {
+  override def configure(): Unit = {
+
+    bind(classOf[AppConfig]).asEagerSingleton()
```

Comment:
I'm assuming that it needs to be eager so that guice can provision it before passing it to other Provides functions  - it might not be the case, I havent had time to test it either way